src/Cache/CacheOperationTypes/CacheOperationTypes.php

```diff
@@ -980,53 +989,49 @@ public function asError(): CacheDictionaryGetFieldsResponseError|null
     }
 }
 
-class CacheDictionaryGetFieldsResponseSuccess extends CacheDictionaryGetFieldsResponse
+class CacheDictionaryGetFieldsResponseHit extends CacheDictionaryGetFieldsResponse
 {
-    private array $responsesList = [];
+    private array $responses = [];
+    private array $valuesDictionary = [];
 
-    public function __construct(_DictionaryGetResponse $responses = null, ?int $numRequested = null)
+    public function __construct(_DictionaryGetResponse $responses, ?array $fields = null)
     {
-        if (!is_null($responses) && is_null($numRequested)) {
-            parent::__construct();
-            foreach ($responses->getFound()->getItems() as $response) {
-                if ($response->getResult() == ECacheResult::Hit) {
-                    $this->responsesList[] = new CacheDictionaryGetFieldResponseHit(null, $response->getCacheBody());
-                }
-                if ($response->getResult() == ECacheResult::Miss) {
-                    $this->responsesList[] = new CacheDictionaryGetFieldResponseMiss();
-                } else {
-                    $this->responsesList[] = new CacheDictionaryGetFieldResponseError(new UnknownError(strval($response->getResult())));
-                }
-            }
-        }
-        if (is_null($responses) && !is_null($numRequested)) {
-            foreach (range(0, $numRequested - 1) as $ignored) {
-                $this->responsesList[] = new CacheDictionaryGetFieldResponseMiss();
+        $counter = 0;
+        parent::__construct();
+        foreach ($responses->getFound()->getItems() as $response) {
+            if ($response->getResult() == ECacheResult::Hit) {
+                $this->responses[] = new CacheDictionaryGetFieldResponseHit(null, $response->getCacheBody());
+                $this->valuesDictionary[$fields[$counter]] = $response->getCacheBody();
+            } else if ($response->getResult() == ECacheResult::Miss) {
```

Comment:
My IDE is telling me that PHP standards prefer `elseif`.
```suggestion
            } elseif ($response->getResult() == ECacheResult::Miss) {
```